client/data/home/use-home-layout-query.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { useQuery, UseQueryResult } from 'react-query';
+import config from '@automattic/calypso-config';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+const useHomeLayoutQuery = (
+	siteId: number | null,
+	isDev?: boolean,
+	forcedView?: string
+): UseQueryResult => {
+	isDev = config.isEnabled( 'home/layout-dev' ) || isDev;
+
+	return useQuery(
+		// Don't include `isDev` or `forcedView` in query key, we want every
+		// caller to have the same idea of what the current view is.
+		[ 'home-layout', siteId ],
```

Comment:
I tried including `isDev` and `forcedView` in the key, but it didn't end up working well. I always want new layouts to replace old layouts, regardless of whether it was originally requested with a forced view or not.

For example, I use `?view=VIEW_PODCASTING` to load a specific view, and then skip that view. When calling the skip endpoint it returns the view to show now, so I replace the old state with `queryClient.setQueryState()`. But in order to overwrite the old state I need to remember the old view ID so that I overwrite the correct cache key. What if I skip again? Would I have to remember the old-old view ID so I can update the cache state?

It also made it difficult to query what the "current view" was. In order to call `queryClient.getQueryState(['home-layout', siteId, ...])` I need to know all parts of the cache key. But now it's chicken and egg: I need the current view ID so that I can look up the layout data and retrieve the current view ID.

> That params changes the meaning of the response

I think that's true from the API's point of view, but from the client's point of view they both mean the same thing: "the one view to show to the user"

Reply:
OK that makes sense. There is only one `[ 'home-layout', siteId ]` response per site and `forcedView` and `isDev` are just debug flags that don't justify a separate key.